Steve McIntyre
steve.mcintyre@linaro.org

I'm currently working on cross-building bits of Debian while
bootstrapping the armhf port. cpio fails to cross-build because of an
invalid redefinition of stat() during the compile. It's not 100% clear
why this only triggers on a cross-build, but this patch makes it work
for me.

Index: cpio-2.11+dfsg/src/filetypes.h
===================================================================
--- cpio-2.11+dfsg.orig/src/filetypes.h
+++ cpio-2.11+dfsg/src/filetypes.h
@@ -82,7 +82,9 @@
 #define lstat stat
 #endif
 
-#ifndef __WIN32__
+#ifndef lstat
 int lstat ();
+#endif
+#ifndef stat
 int stat ();
 #endif
